rs/tests/boundary_nodes/rate_limit_canister_test.rs

```diff
@@ -1,403 +1,303 @@
-                rule_raw: rule_4.to_bytes_json().unwrap(),
-                description: "Some vulnerability #3 discovered".to_string(),
-            },
-        ],
-    })
-    .unwrap();
+    info!(
+        &logger,
+        "Step 5. Verify that agent can successfully interact with counter canister"
+    );
 
-    let result = agent
-        .update(&canister_id, "add_config")
-        .with_arg(args)
+    api_bn_agent
+        .update(&counter_canister_id, "write")
         .call_and_wait()
         .await
-        .unwrap();
+        .expect("failed to increment the counter on canister");
 
-    let decoded = Decode!(&result, AddConfigResponse).unwrap();
+    info!(
+        &logger,
+        "Step 6. Add a rate-limit rule that blocks requests to counter canister"
+    );
 
-    info!(&logger, "Response to add_config() call: {decoded:#?}");
-}
+    set_rate_limit_rule(
+        &api_bn_agent,
+        rate_limit_id,
+        RateLimitRule {
+            canister_id: Some(counter_canister_id),
+            subnet_id: None,
+            methods_regex: None,
+            request_types: None,
+            limit: Action::Block,
+        },
+    )
+    .await;
 
-async fn read_config(
-    logger: Logger,
-    agent: &Agent,
-    version: Version,
-    canister_id: Principal,
-) -> Vec<RuleId> {
-    let args = Encode!(&Some(version)).unwrap();
+    info!(
+        &logger,
+        "Step 7. Verify that the api_bn_agent can no longer send requests to the counter canister"
+    );
 
-    let response = agent
-        .query(&canister_id, "get_config")
-        .with_arg(args)
-        .call()
-        .await
-        .expect("update call failed");
+    retry_with_msg_async!(
+        "check_counter_canister_becomes_unreachable".to_string(),
+        &logger,
+        Duration::from_secs(180),
+        Duration::from_secs(5),
+        || async {
+            match timeout(
+                Duration::from_secs(2),
+                api_bn_agent.update(&counter_canister_id, "write").call(),
+            )
+            .await
+            {
+                Ok(_) => bail!("counter canister is still reachable, retrying"),
+                Err(_) => Ok(()),
+            }
+        }
+    )
+    .await
+    .expect("failed to check that canister becomes unreachable");
 
-    let decoded = Decode!(&response, GetConfigResponse)
-        .expect("failed to decode candid response")
-        .unwrap();
+    info!(
+        &logger,
+        "Step 8. Add a rate-limit rule, which unblocks requests to the counter canister"
+    );
 
-    info!(&logger, "Response to get_config() call: {}", decoded);
+    set_rate_limit_rule(
+        &api_bn_agent,
+        rate_limit_id,
+        RateLimitRule {
+            canister_id: Some(counter_canister_id),
+            subnet_id: None,
+            methods_regex: None,
+            request_types: None,
+            limit: Action::Limit(300, Duration::from_secs(60)),
+        },
+    )
+    .await;
```

Comment:
What's the reasoning for adding another rule instead of just removing it?

Reply:
With this approach we check a bit more scenarios on the` ic-boundary` side:
- no blocking rule present - sending requests works
- blocking rule present - sending requests doesn't work
- relaxed rate-limit rule present - sending requests also works
That was the rationale.